### DISCUSSION TITLE
Randomization with submit button in between

### DISCUSSION THREAD

**Discussion by fidadoma** [rubenarslan/formr.org#553] [Q&A] [answered]
Dear all,
we are preparing study, in which we want to show randomized blocks (easy), however some blocks are divided into two halves and if we randomize, we want to make sure that the items are randomized for whole block, while mid-submit button stays the same. 
In particular, it looks like following:

![image](https://github.com/rubenarslan/formr.org/assets/3858193/79b222b4-b274-4249-94bf-fb352729bd83)

so either aa or cc should be shown, but when aa block is shown, we should first show two items on first page and remaing two on the second page.

I thought about using R code in item_order column, but it did not work and it seems that I can't set these values dynamically. What is the best way to achieve this? (if possible)

Thanks
Filip

**Comment by rubenarslan** [comment]
I'm not sure I correctly understand the goal. Items are first ordered by block, then by item order. So you can repeat the same item order in each block (e.g. 2 (note 1),  2 (note 2), 3 (submit). Then the submit button is always last in each block, but the items in each block are in randomized order.

**Reply by fidadoma** [reply]
sorry for not making myself clear- I can make it to randomly sample note 1, note2 and then show submit, but I want to first randomly show two items out of note1 through 4, then submit button, then the remaining two items out of note1 throug 4. So basically all four items would have same item order, but the middle submit button should be fixed in the middle. Is this possible somehow?  thank you

**Reply by rubenarslan** [reply, marked as answer]
Ah, I'm afraid I have no __good__ idea how to do this. You can go the very involved route of manually creating your item order in R and filling in the item labels, but this is a pain to implement and analyse later: https://docs.google.com/spreadsheets/d/1jkts0xp1JmFX5t1A34kLswH7ZyWWn8xArKeCH2-pzH4/edit#gid=2037039862

Making it possible to write R code to define the item order definitely seems like the superior solution, but it's rarely been requested.

**Reply by fidadoma** [reply]
Ok, we omitted the randomization, thanks!
